cpp/example_code/autoscaling/tests/autoscaling_gtests.cpp

```diff
@@ -5,11 +5,63 @@
 #include <fstream>
 #include <aws/core/client/ClientConfiguration.h>
 #include <aws/testing/mocks/http/MockHttpClient.h>
+
 static const char ALLOCATION_TAG[] = "AUTOSCALING_GTEST";
 
 Aws::SDKOptions AwsDocTest::AutoScaling_GTests::s_options;
 std::unique_ptr<Aws::Client::ClientConfiguration> AwsDocTest::AutoScaling_GTests::s_clientConfig;
 
+/*
+ * Subclass MockHTTPCLient to respond to credential requests.
+ * Otherwise, the stored responses are returned for credential requests
+ * and not the service API calls.
+ */
+class CustomMockHTTPClient : public MockHttpClient {
+public:
+    explicit CustomMockHTTPClient(
+            const std::shared_ptr<Aws::Http::HttpRequest> &requestTmp) {
+        std::shared_ptr<Aws::Http::Standard::StandardHttpResponse> goodResponse = Aws::MakeShared<Aws::Http::Standard::StandardHttpResponse>(
+                ALLOCATION_TAG, requestTmp);
+        goodResponse->AddHeader("Content-Type", "text/json");
+        goodResponse->SetResponseCode(Aws::Http::HttpResponseCode::OK);
+        Aws::Utils::DateTime expiration =
+                Aws::Utils::DateTime::Now() + std::chrono::milliseconds(60000);
+
+        goodResponse->GetResponseBody() << R"({
+"RoleArn":"arn:aws:iam::123456789012:role/MockRole",
+"AccessKeyId":"ABCDEFGHIJK",
+"SecretAccessKey":"ABCDEFGHIJK",
+"Token":"ABCDEFGHIJK==",
+"Expiration":")"
+                                        << expiration.ToGmtString(
+                                                Aws::Utils::DateFormat::ISO_8601)
+                                        << R"("
+    })";
+        this->AddResponseToReturn(goodResponse);
```

Comment:
(nit) IMHO the embedded newlines reduce readability of the C++ and are unnecessary for the JSON.

```
        goodResponse->GetResponseBody() << "{"
            << R"("RoleArn":"arn:aws:iam::123456789012:role/MockRole",)"
            << R"("AccessKeyId":"ABCDEFGHIJK",)"
            << R"("SecretAccessKey":"ABCDEFGHIJK",)"
            << R"(Token":"ABCDEFGHIJK==","Expiration":")" << expiration.ToGmtString(Aws::Utils::DateFormat::ISO_8601) << R"(")"
            << "}";
        this->AddResponseToReturn(goodResponse);
```